### PR TITLE
Add support for path and configuration in dependency block

### PR DESCRIPTION
**Pull request by joshafeinberg**

### CODE REVIEW

File: shadowed/antlr/src/main/antlr/com/autonomousapps/internal/grammar/gen/GradleGroovyScriptLexer.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Generated from /Users/trobalik/Development/dependency-analysis-android-gradle-plugin/shadowed/antlr/src/main/antlr/com/autonomousapps/internal/grammar/GradleGroovyScriptLexer.g4 by ANTLR 4.10.1
+// Generated from /Users/joshfeinberg/projects/dependency-analysis-android-gradle-plugin/shadowed/antlr/src/main/antlr/com/autonomousapps/internal/grammar/GradleGroovyScriptLexer.g4 by ANTLR 4.10.1
```
Comment (joshafeinberg) [inline shadowed/antlr/src/main/antlr/com/autonomousapps/internal/grammar/gen/GradleGroovyScriptLexer.java:1]: Do we want to not commit this generated code? Not sure if it makes sense to add this to .gitignore

Comment (autonomousapps) [reply shadowed/antlr/src/main/antlr/com/autonomousapps/internal/grammar/gen/GradleGroovyScriptLexer.java:1]: We absolutely must commit this generated code. It must go into the compiled jar that is published on maven central.

File: shadowed/antlr/src/test/groovy/com/autonomousapps/GradleGroovyScriptSpec.groovy [hunk 2]
```diff
@@ -47,6 +47,8 @@ final class GradleGroovyScriptSpec extends Specification {
       dependencies {
         implementation 'heart:of-gold:1.0'
         api project(":marvin")
+        api project(path: ':deep-thought')
+        api project(path: ':infinite-improbability-drive', configuration: 'withoutOsgi')
```
Comment (autonomousapps) [inline shadowed/antlr/src/test/groovy/com/autonomousapps/GradleGroovyScriptSpec.groovy:51]: I appreciate your commitment to the bit :)

**Review by autonomousapps — COMMENTED** [review]
Let's see what the test suite says! Thanks for the submission.

**Comment by joshafeinberg** [comment]
Hey, just wanted to bump this as we're hoping to begin using your plugin next year. Know we chatted a bit elsewhere but let me know if theres anything I can do to move this

**Comment by autonomousapps** [comment]
@joshafeinberg really sorry about this quite long delay. The latter part of the year really hammered me at work. This PR has been on my mind 😅 No promises, but I will do my best to get to it soon ™️

**Comment by autonomousapps** [comment]
Sorry for the very long delay here, but a recent [merge to main](https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/940) should have fixed this!

**Pull request closed without merging**